RIOT-OS/RIOT
# make: pass make jobserver to cargo

### Contribution description

For cargo to honour our make's jobserver, it's recipe line needs to be prefixed with `+`.
Make >= 4.4 currently also needs the `--jobserver-style=pipe` argument. (Our CI is using 4.3).



### Testing procedure

`make -Ctests/rust-hello-world clean all -j2`

On master, observe as many rustc processes (e.g., in top) as there are CPUs. With this PR, there should only start two.

(should your system have make 4.4, try `make --jobserver-style=pipe -Ctests/rust-hello-world clean all -j2`)




### Issues/PRs references

Fixes https://github.com/RIOT-OS/RIOT/issues/19045.
